units/recalculate_change_points_test.go

```diff
@@ -129,15 +134,47 @@ func makePerfResultsWithChangePoints(unique string, seed int64) ([]testResultsAn
 	}
 
 	// time to map over change points
-	changePoints := map[string][]int{}
+	changePoints := map[string][]changePointIndexPercentChange{}
 	for measurement, cpIndices := range timeSeriesChangePoints {
 		measurementName := "measurement_" + strconv.Itoa(measurement)
-		changePoints[measurementName] = append(changePoints[measurementName], cpIndices...)
+		cpsWithIndexAndPercentChange := make([]changePointIndexPercentChange, len(cpIndices))
+		currentTimeSeries := timeSeries[measurement]
+		for i, pointIndex := range cpIndices {
+			var lowerBound int
+			var upperBound int
+
+			if i == 0 {
+				lowerBound = 0
+			} else {
+				lowerBound = cpIndices[i-1]
+			}
+
+			if i == len(cpIndices)-1 {
+				upperBound = len(currentTimeSeries)
+			} else {
+				upperBound = cpIndices[i+1]
+			}
+
+			lowerWindow := convertIntArrayToFloat64(currentTimeSeries[lowerBound:pointIndex])
+			upperWindow := convertIntArrayToFloat64(currentTimeSeries[pointIndex:upperBound])
+
+			percentChange := calculatePercentChange(lowerWindow, upperWindow)
+			cpsWithIndexAndPercentChange[i] = changePointIndexPercentChange{index: pointIndex, percentChange: percentChange}
+		}
+		changePoints[measurementName] = append(changePoints[measurementName], cpsWithIndexAndPercentChange...)
```

Comment:
This will work. One thing to note is that we are basically checking that we get the percent change in the same way the job itself gets the percent change. It's possible that our average calculations are bugged though, and this test would pass (since it uses the same procedure).

Should we change this to be more of a sanity check? Our test data's stable regions are constant/have no noise, and cps have a minimum spacing in the test, so we could do something like:
```
# No need to average over windows, since they are constant --> just use points before/after the change
percentChange := 100*((currentTimeSeries[pointIndex]/currentTimeSeries[pointIndex-1])-1)
cpsWithIndexAndPercentChange[i] = changePointIndexPercentChange{index: pointIndex, percentChange: percentChange}
```
And drop all of the windowing stuff.

What do you think?

Reply:
Seems good to me, since the regions are stable. I've added your change